### ISSUE TITLE
Error in husky install

### ISSUE THREAD

**Issue by dantefff** [dionysius/vaultwarden-web-vault-deb#1]
I'm trying to make bullseye version of your latest tag, and can't compile.

I have node 18 installed with nodesource.

**This is what I do:**
```
gbp clone https://github.com/dionysius/vaultwarden-web-vault-deb.git
cd vaultwarden-deb
git reset --hard $(git describe --abbrev=0)
gbp buildpackage -d -b --git-export-dir=. --git-ignore-new -us -uc
```
**And this is what I get:**
```
gbp:info: Creating vaultwarden-web-vault_2023.5.0.orig.tar.gz from 'web-v2023.5.0'
gbp:info: Extracting 'vaultwarden-web-vault_2023.5.0.orig.tar.gz' to '/home/noone/vaultwarden/vaultwarden-web-vault-deb/package/vaultwarden-web-vault-tmp'
gbp:info: Exporting 'HEAD' to '/home/noone/vaultwarden/vaultwarden-web-vault-deb/package/vaultwarden-web-vault-tmp'
gbp:info: Moving '/home/noone/vaultwarden/vaultwarden-web-vault-deb/package/vaultwarden-web-vault-tmp' to '/home/noone/vaultwarden/vaultwarden-web-vault-deb/package/vaultwarden-web-vault-2023.5.0'
gbp:info: Performing the build
 dpkg-buildpackage -us -uc -ui -i -I -d -b
dpkg-buildpackage: info: source package vaultwarden-web-vault
dpkg-buildpackage: info: source version 2023.5.0-2
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by dionysius <dragon.dionysius@gmail.com>
 dpkg-source -i -I --before-build .
dpkg-buildpackage: info: host architecture amd64
dpkg-source: info: using options from vaultwarden-web-vault-2023.5.0/debian/source/options: --extend-diff-ignore=(^|/)(.github|LICENSE|README|signing-key)
dpkg-source: info: using patch list from debian/patches/series
dpkg-source: info: applying web_vault.patch
 debian/rules clean
dh clean
   dh_clean
        rm -f debian/debhelper-build-stamp
        rm -rf debian/.debhelper/
        rm -f -- debian/vaultwarden-web-vault.substvars debian/files
        rm -fr -- debian/vaultwarden-web-vault/ debian/tmp/
        find .  \( \( \
                \( -path .\*/.git -o -path .\*/.svn -o -path .\*/.bzr -o -path .\*/.hg -o -path .\*/CVS -o -path .\*/.pc -o -path .\*/_darcs \) -prune -o -type f -a \
                \( -name '#*#' -o -name '.*~' -o -name '*~' -o -name DEADJOE \
                 -o -name '*.orig' -o -name '*.rej' -o -name '*.bak' \
                 -o -name '.*.orig' -o -name .*.rej -o -name '.SUMS' \
                 -o -name TAGS -o \( -path '*/.deps/*' -a -name '*.P' \) \
                \) -exec rm -f {} + \) -o \
                \( -type d -a -name autom4te.cache -prune -exec rm -rf {} + \) \)
 debian/rules binary
dh binary
   dh_update_autotools_config
   dh_autoreconf
   debian/rules override_dh_auto_configure
cp -rf debian/patches/resources/* apps/web/
   debian/rules override_dh_auto_build
npm ci
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated trim@0.0.1: Use String.prototype.trim() instead
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated asar@3.2.0: Please use @electron/asar moving forward.  There is no API change, just a package name change
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated electron-osx-sign@0.6.0: Please use @electron/osx-sign moving forward. Be aware the API is slightly different
npm WARN deprecated dommatrix@1.0.3: dommatrix is no longer maintained. Please use @thednp/dommatrix.
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.

> @bitwarden/clients@0.0.0 prepare
> husky install

husky - .git can't be found (see https://typicode.github.io/husky/#/?id=custom-directory)
npm ERR! code 1
npm ERR! path /home/noone/vaultwarden/vaultwarden-web-vault-deb/package/vaultwarden-web-vault-2023.5.0
npm ERR! command failed
npm ERR! command sh -c husky install

npm ERR! A complete log of this run can be found in: /home/noone/.npm/_logs/2023-08-05T18_04_35_497Z-debug-0.log
make[1]: *** [debian/rules:24: override_dh_auto_build] Error 1
make: *** [debian/rules:16: binary] Error 2
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
debuild: fatal error at line 1182:
dpkg-buildpackage -us -uc -ui -i -I -d -b failed
gbp:error: 'debuild -i -I -d -b -us -uc' failed: it exited with 29

```

Seems that husky must have a .git in the same folder as package.json is or at least have a cd.. and a install in a subfolder (see https://stackoverflow.com/questions/74129312/how-to-configure-husky-when-git-is-in-a-different-folder)

**Comment by dionysius** [comment]
I'm on holiday so lets see when I can look into it. In the meantime you should be able to just use the .deb of the bookworm build as the resulting package is arch independent and technically also debian version independent - it just contains the resulting webpage files

**Comment by dionysius** [comment]
according to CI I'm using node 16.x: https://github.com/dionysius/vaultwarden-web-vault-deb/blob/19adb5c19f3afc78511df5cd7b436683c3a97ebc/.github/workflows/packaging.yml#L22

Also the vaultwarden patching repo still uses node 16.x: https://github.com/dani-garcia/bw_web_builds/blob/master/Dockerfile#L20

And this repo is recreating what they are using, so I guess give node 16 a try. You can use nvm to make additional installations easier managable

**Comment by dionysius** [comment]
Also the official patching repo still uses node 16.x: https://github.com/dani-garcia/bw_web_builds/blob/master/Dockerfile#L20

**Comment by dantefff** [comment]
Thanks a lot for answering. There is no rush at all. I have used the bookworm deb as you mentioned without any problems. And I have been able to build the bullseye version of vaultwarden-deb.

I will take a look at node16 and I will come back to let you know the results.

Thanks a lot again for your work.

**Comment by dionysius** [comment]
I'll look into it as well and probably update the readme a bit. To be fair the nvmrc file on the official upstream source seems to use node 18. I suspect 16 won't help but I hope i find out more

**Comment by dionysius** [comment]
Well looks like I found the cause. I spun up a fresh lxc bullseye container and dug around. It seems it doesn't like it if you tell it to export it to the current folder. I get your error if I use `--git-export-dir=.`, but works fine if I use `--git-export-dir=..`.

I still get the following warning, slightly different, but it continues:
```
> @bitwarden/clients@0.0.0 prepare
> husky install

fatal: not a git repository (or any of the parent directories): .git
husky - git command not found, skipping install

added 3528 packages, and audited 3540 packages in 1m
```

It kinda makes sense, as gbp has two branches with data that makes the debian source package, the debian branch and the upstream branch. It needs to overlay the data from both branches and thus should do that in a clean folder. It also ignores/cleans up some data during the overlay, such as .git folders. Since husky now found your .git folder, it tried to do stuff. So beeing allowed to export it to the current folder depends on the source. Also note that commands not referred in build-dependencies are usually also not available during the build, this is probably why `git` is not available and I got "lucky" the build continues. husky is anyway unnecessary for the build - if it starts to hinder the build more, i can write a patch for it that is automatically applied onto the source during build.

Note to self:
- [x] ~~nvm 18 did not work, 16 was fine, so something is off with the workaround~~
- [x] bookworm has 18 already available, could skip installing nodesource during CI
- [x] I should update ci to 18, as upstream .nvmrc is suggesting to build with 18
